lib/modules/manager/pip_requirements/artifacts.spec.ts

```diff
@@ -54,53 +65,81 @@ describe('modules/manager/pip_requirements/artifacts', () => {
     ).toBeNull();
   });
 
-  it('returns null if unchanged', async () => {
-    fs.readLocalFile.mockResolvedValueOnce(newPackageFileContent);
-    const execSnapshots = mockExecAll();
-    expect(
-      await updateArtifacts({
-        packageFileName: 'requirements.txt',
-        updatedDeps: [{ depName: 'atomicwrites' }],
-        newPackageFileContent,
-        config,
-      })
-    ).toBeNull();
-
-    expect(execSnapshots).toMatchObject([
-      {
-        cmd: 'hashin atomicwrites==1.4.0 -r requirements.txt',
-        options: { cwd: '/tmp/github/some/repo' },
-      },
-    ]);
-  });
-
-  it('returns updated file', async () => {
-    fs.readLocalFile.mockResolvedValueOnce('new content');
-    const execSnapshots = mockExecAll();
-    expect(
-      await updateArtifacts({
-        packageFileName: 'requirements.txt',
-        updatedDeps: [{ depName: 'atomicwrites' }],
-        newPackageFileContent,
-        config,
-      })
-    ).toEqual([
-      {
-        file: {
-          type: 'addition',
-          path: 'requirements.txt',
-          contents: 'new content',
+  it.each([
+    ['dependency w/o extras', 'atomicwrites', 'atomicwrites==1.4.0'],
+    [
+      'dependency with extras',
+      'boto3-stubs',
+      'boto3-stubs[iam] == 1.24.36.post1',
+    ],
+  ])(
```

Comment:
please use a single test and pass all values as `updatedDeps`. So that `updateArtifacts` should update all hashes , so you get three exec snapshots

Reply:
Now you say it... Makes sense, of course. Done.